src/Map.php

```diff
@@ -241,7 +241,9 @@ private function getHash($key)
             // Take a hash of the serialized value to avoid PHP having to do a
             // long string comparison on each lookup (we already check the
             // actual array values for equality ourselves).
-            return md5(serialize($key));
+
+            // Sometimes serializing the whole array fails because the array contains anonymous classes or closures.
+            return md5(serialize(array_map(fn($val) => $this->getHash($val), $key)));
```

Comment:
There's a fair amount going on here. I might separate the recursive hash step from the serialize + md5 step.